WordPress/Classes/Services/PostService.m

```diff
@@ -206,7 +221,7 @@ - (void)syncPostsOfType:(NSString *)postType
                    success:^(NSArray *posts) {
                        BOOL hasMore = ([posts count] < PostServiceNumberToFetch) ? NO : YES;
                        [self.managedObjectContext performBlock:^{
-                           [self mergePosts:posts ofType:postType withStatuses:postStatus forBlog:blog purgeExisting:YES completionHandler:^{
+                           [self mergePosts:posts ofType:postType withStatuses:postStatus byAuthor:authorID forBlog:blog purgeExisting:YES completionHandler:^{
```

Comment:
> Syncing is hard

I think we make it harder for ourselves. For example having a way to cancel an operation in progress would address the scenario you described but afaik there's no way to do it with our "fire and forget" approach to API calls. 

> never get around to fix the actual problem

We've got some new API for the reader for addressing gaps. If it works well we'll likely apply the same logic to site endpoints.